File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, ans = INT_MIN;
        for(int i: nums){
            sum += i;
            ans = max(ans, sum);
            if(sum<0)
                sum = 0;
        }
        return ans;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxiSum = Integer.MIN_VALUE, currSum = 0;
        for(int i = 0; i<nums.length; i++){
            currSum = currSum+nums[i];
            maxiSum = Math.max(maxiSum, currSum);
            currSum = Math.max(currSum, 0);
        }
        return maxiSum;
    }
}